src/classes/FeatureParameterMapper.cls

```diff
@@ -63,7 +109,71 @@ public virtual with sharing class FeatureParameterMapper {
     public virtual FeatureParameterModel getFeatureParameterModel(String featureParameter) {
-        FeatureParameterDateModel featureParamDateMdl = new FeatureParameterDateModel(featureParameter, Date.today());
-        return featureParamDateMdl;
+
+        TelemetryParameterName lastRunDate = this.convertToEnum(featureParameter);
+
+        if (lastRunDate == null) {
+            throw new FeatureParameterMapperException('An error in Feature Parameter');
+        }
+
+        String featureParameterDataType = this.getFeatureParameterType().get(lastRunDate);
+
+        if (featureParameterDataType == 'Date') {
+            UTIL_FeatureManagement featureManager = new UTIL_FeatureManagement();
+            Date settingsLastRunDate = featureManager.checkPackageDateValue(String.valueOf(lastRunDate));
+            FeatureParameterDateModel featureParamDateMdl = new FeatureParameterDateModel(String.valueOf(lastRunDate), settingsLastRunDate);
+            return featureParamDateMdl;
+        } 
+
+        throw new FeatureParameterMapperException('An error in Feature Parameter');
+
     }
+
+    /*************************************************************************************************************
+    * @description Updates the feature parameter's value
+    * @Params featureParameter The feature Parameter to update the parameter from
+    **************************************************************************************************************/
+    public virtual void updateFromFeatureParameterModel(FeatureParameterModel featureParamModel) {
+
+        TelemetryParameterName lastRunDate = this.convertToEnum(featureParamModel.name);
+
+        if (lastRunDate == null) {
+            throw new FeatureParameterMapperException('An error in Feature Parameter');
+        }
+
+        String dataType = this.getFeatureParameterType().get(lastRunDate);
+
+        if (dataType == 'Date') {
+            FeatureParameterDateModel featureParamDateModel = (FeatureParameterDateModel)featureParamModel;
+            UTIL_FeatureManagement featureManager = new UTIL_FeatureManagement();
+            featureManager.setPackageDateValue(featureParamDateModel.name, featureParamDateModel.value);
+        } else {
+            throw new FeatureParameterMapperException('An error in Feature Parameter1');
+        }
+
+    }
+
+    /*************************************************************************************************************
+    * @description Returns the enum equivalent of a String if found else returns null
+    * @Params toBeConvertedString The String that needs to be converted to Enum
+    * @return TelemetryParameterName enum equivalent of the String passed
+    **************************************************************************************************************/
+    @TestVisible
+    private TelemetryParameterName convertToEnum(String toBeConvertedString) {
```

Comment:
@MatthewBlanski slightly updated the conversion and error handling w.r.t to the SPEC, wanted to check with you on this if this is ok?